File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Commons/CategorySchema.cs
using System;
using System.Collections.Generic;

namespace VecompSoftware.DocSuiteWeb.Entity.Commons
{
    public class CategorySchema : DSWBaseEntity
    {
        #region [ Constructor ]

        public CategorySchema() : this(Guid.NewGuid()) { }
        public CategorySchema(Guid uniqueId)
            : base(uniqueId)
        {
            Categories = new HashSet<Category>();
        }
        #endregion

        #region[ Properties ]

        public short Version { get; set; }
        public DateTimeOffset StartDate { get; set; }
        public DateTimeOffset? EndDate { get; set; }
        public string Note { get; set; }

        #endregion

        #region [ Navigation Properties ]

        public virtual ICollection<Category> Categories { get; set; }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Common/CategorySchema.vb
Imports Newtonsoft.Json
Imports System.Text

<Serializable()>
Partial Public Class CategorySchema
    Inherits AuditableDomainObject(Of Guid)

    Public Sub New()
        MyBase.Id = Guid.NewGuid()
    End Sub

    Public Overridable Property Version As Short
    Public Overridable Property StartDate As DateTimeOffset
    Public Overridable Property EndDate As DateTimeOffset?
    Public Overridable Property Note As String
    Public Overridable Property Categories As IList(Of Category)


End Class

